src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CustomiseHearingBundleHandler.java

```diff
@@ -0,0 +1,281 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.DateProvider;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.*;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.em.Bundle;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.Appender;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.BundleRequestExecutor;
+
+
+@Component
+public class CustomiseHearingBundleHandler  implements PreSubmitCallbackHandler<AsylumCase> {
+    private final BundleRequestExecutor bundleRequestExecutor;
+    private final Appender<DocumentWithMetadata> appender;
+    private final DateProvider dateProvider;
+    private final String emBundlerUrl;
+    private final String emBundlerStitchUri;
+
+    public CustomiseHearingBundleHandler(
+            @Value("${emBundler.url}") String emBundlerUrl,
+            @Value("${emBundler.stitch.uri}") String emBundlerStitchUri,
+            BundleRequestExecutor bundleRequestExecutor,
+            Appender<DocumentWithMetadata> appender,
+            DateProvider dateProvider
+    ) {
+        this.emBundlerUrl = emBundlerUrl;
+        this.emBundlerStitchUri = emBundlerStitchUri;
+        this.bundleRequestExecutor = bundleRequestExecutor;
+        this.appender = appender;
+        this.dateProvider = dateProvider;
+
+    }
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == Event.CUSTOMISE_HEARING_BUNDLE;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+                callback
+                        .getCaseDetails()
+                        .getCaseData();
+
+        prepareDocuments(asylumCase, CUSTOM_HEARING_DOCUMENTS,HEARING_DOCUMENTS);
```

Comment:
I would rather create deep-copy of `asylumCase` for example `asylumCaseCopy` by using Jackson mapping example below:
https://www.baeldung.com/java-deep-copy
and pass to `prepareDocuments` method

The original `asylumCase` could be used as a return response object after adding `CASE_BUNDLES` and `STITCHING_STATUS` fields

We can't use caseDetailsBefore for that purpose.